src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -1840,31 +1841,94 @@ void WorldSession::HandleCharFactionOrRaceChangeCallback(std::shared_ptr<Charact
+                TaxiMask SwitchedTaxiMask;
+                memset(&SwitchedTaxiMask, 0, sizeof(SwitchedTaxiMask));
 
                 TaxiMask const& factionMask = newTeam == HORDE ? sHordeTaxiNodesMask : sAllianceTaxiNodesMask;
-                for (uint8 i = 0; i < numFullTaximasks; ++i)
+                for (auto const& itr : sTaxiPathSetBySource)
                 {
-                    uint8 deathKnightExtraNode = (playerClass == CLASS_DEATH_KNIGHT) ? sDeathKnightTaxiNodesMask[i] : 0;
-                    taximaskstream << uint32(factionMask[i] | deathKnightExtraNode) << ' ';
+                    auto TaxiMaskBuild = [&](uint8 field, uint32 mask)
+                    {
+                        if (playerClass == CLASS_DEATH_KNIGHT)
+                        {
+                            SwitchedTaxiMask[field] |= uint32(mask | (sDeathKnightTaxiNodesMask[field] & mask));
+                        }
+                        else
+                        {
+                            SwitchedTaxiMask[field] |= mask;
+                        }
+                    };
+
+                    uint32 nodeId = itr.first;
+                    uint8 field = (uint8)((nodeId - 1) / 32);
+                    uint32 submask = 1 << ((nodeId - 1) % 32);
+
+                    if ((factionMask[field] & submask) == 0)
+                    {
+                        TaxiMaskBuild(field, 0);
+                        continue;
+                    }
+
+                    TaxiPathSetForSource const& taxiPaths = itr.second;
+                    if (taxiPaths.empty())
+                    {
+                        TaxiMaskBuild(field, 0);
+                        continue;
+                    }
+                    TaxiPathEntry const* taxiPath = taxiPaths.begin()->second;
+                    if (!taxiPath)
+                    {
+                        TaxiMaskBuild(field, 0);
+                        continue;
+                    }
+
+                    TaxiPathNodeList const& taxiNodePaths = sTaxiPathNodesByPath[taxiPath->ID];
+                    if (taxiNodePaths.empty())
+                    {
+                        TaxiMaskBuild(field, 0);
+                        continue;
+                    }
+
+                    TaxiPathNodeEntry const* pathNode = taxiNodePaths.front();
+                    if (!pathNode)
+                    {
+                        TaxiMaskBuild(field, 0);
+                        continue;
+                    }
+
+                    AreaTableEntry const* zone = sAreaTableStore.LookupEntry(sMapMgr->GetZoneId(PHASEMASK_NORMAL, pathNode->ContinentID, pathNode->Loc.X, pathNode->Loc.Y, pathNode->Loc.Z));
+                    if (!zone)
+                    {
+                        TaxiMaskBuild(field, 0);
+                        continue;
+                    }
+
+                    LFGDungeonEntry const* lfgZone = GetLFGZoneEntry(zone->AreaName[GetSessionDbLocaleIndex()], GetSessionDbLocaleIndex());
```

Comment:
ref my earlier comment - you already have AreaTable data right here, why bother looking at lfg when you can just read ExplorationLevel from it?

Reply:
I can try to re-explore it when i get back home but i had a issue with needing to read the dbc store portion to pull it. Unless what i attempted was just not the propper way.

Reply:
@Shauren ok. i attempted and i still come down to the same results. the AreaTable will not work. So looking at the dbc file, we would have to go off explorationlevel collum. however, we got huge discrepencies. over 800 spots have the exploration level as either -1 or -, it becomes a mess of sort, and nonsensible in game, we will end up with flight points of other continents outside of the intended player level, such as parts of northerend being unlocked for low level players.
This is the snippet from the dbc
![image](https://user-images.githubusercontent.com/16887899/163072610-4dbd8418-8fc2-4e74-ade2-3fc87e7eebb0.png)

Now my choice was to go with the min level collumn of the LFGDungeons.dbc which provides a more appropiate outcome for the end user due to minal level and area name being used.
This is the snippet from that dbc
![image](https://user-images.githubusercontent.com/16887899/163072847-6f673be3-1b6d-42ee-bad5-b2362e56110d.png)

At this point as much as i do want to honor your request and review and change it, i just simply can not as the results will not be the appropiate outcome. I guess all i can ask at this point, is maybe discuss it internal for a better way in handling it, but also understand i will not take any offense to the pr being closed. I know with developement there are always another way to approach the subject and the intended goal. However at my age of boomerness, i am pretty much feeling capped off with what i can do.